pmlc/conversion/stdx_to_spirv/stdx_to_spirv.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020, Intel Corporation
+
+#include "mlir/Dialect/SPIRV/SPIRVDialect.h"
+#include "mlir/Dialect/SPIRV/SPIRVOps.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "pmlc/conversion/stdx_to_spirv/pass_detail.h"
+#include "pmlc/conversion/stdx_to_spirv/passes.h"
+#include "pmlc/dialect/stdx/ir/ops.h"
+#include "pmlc/util/logging.h"
+
+using namespace mlir; // NOLINT[build/namespaces]
+
+namespace pmlc::conversion::stdx_to_spirv {
+
+namespace stdx = dialect::stdx;
+
+namespace {
+struct StdxUitoFOpConversion : public OpConversionPattern<stdx::UIToFPOp> {
+  using OpConversionPattern<stdx::UIToFPOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(stdx::UIToFPOp uitofOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    TypeConverter typeConverter;
+    assert(operands.size() == 1);
+    auto srcType = operands.front().getType();
+    auto dstType = typeConverter.convertType(uitofOp.getResult().getType());
+
+    if (srcType == dstType) {
+      IVLOG(1, "Converting the same type");
+      rewriter.replaceOp(uitofOp, operands.front());
+    } else {
+      /* TODO: Check the corresponding type */
+      auto dstSType = rewriter.getF32Type();
+      auto loc = uitofOp.getLoc();
+      auto convertOp = rewriter.create<spirv::UConvertOp>(
+          loc, dstSType, operands, ArrayRef<NamedAttribute>());
+      rewriter.replaceOpWithNewOp<spirv::ConvertSToFOp>(uitofOp, dstType,
+                                                        convertOp.getResult());
+    }
+    return success();
+  }
+};
+
+struct StdxRoundOpConversion : public OpConversionPattern<stdx::RoundOp> {
+  using OpConversionPattern<stdx::RoundOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(stdx::RoundOp roundOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    TypeConverter typeConverter;
+    assert(operands.size() == 1);
+    /* TODO: Check the corresponding type */
+    auto dstType = rewriter.getF32Type();
+    rewriter.replaceOpWithNewOp<spirv::GLSLCeilOp>(roundOp, dstType,
+                                                   operands.front());
+
+    return success();
+  }
+};
+
+struct LowerStdxToSPIRVPass
+    : public LowerStdxToSPIRVBase<LowerStdxToSPIRVPass> {
+  // Run the dialect converter on the module.
```

Comment:
It needs to override virtual void getDependentDialects(DialectRegistry &registry) const {}.

Reply:
Will do.

Reply:
Please add generated dialects to registry.